### DISCUSSION TITLE
Rust Analyzer won't show diagnostics

### DISCUSSION THREAD

**Discussion by luongngocminh** [LazyVim/LazyVim#2573] [Q&A] [answered]
While `cargo check` would resulted in multiple error.
VsCode is working fine.

cargo check:
<img width="981" alt="image" src="https://github.com/LazyVim/LazyVim/assets/9526028/1a184a26-6d4e-4c2e-91fe-93c5b65b0a7e">

VSCode:
<img width="1112" alt="image" src="https://github.com/LazyVim/LazyVim/assets/9526028/644c0d44-8e79-49b3-9e91-445358737c1e">

Lazyvim:
<img width="894" alt="image" src="https://github.com/LazyVim/LazyVim/assets/9526028/66d25a44-92cc-429a-9348-8fb7620d29b8">
As you can see the LSP server is running with `gd` and type symbol working.
Anyone experiencing this?

**Comment by dpetka2001** [comment]
Try to install Rust via `rustup` like the official [docs](https://www.rust-lang.org/tools/install) describe and see if you get the same behavior.

**Reply by luongngocminh** [reply]
Thanks for helping but why is this related?

**Reply by luongngocminh** [reply]
I've tried it, the problem still persists

**Reply by dpetka2001** [reply]
There was a similar issue with another user using Gentoo and his Rust installation from `emerge` didn't work correctly and ended up installing it via `rustup` and it worked. 

Another user had a similar problem in docker and while I was debugging it I saw that he installed Rust with `apk` (he was using Alpine image distro) and as soon as I replaced that with Rust installed via `rustup` it worked.

I'm not familiar with how the package managers of Linux distros create the packages and how that affects the builtin Neovim LSP, but I've seen it twice so far. That's why I suggested you tried `rustup` and see if that solves your problem.

**Reply by dpetka2001** [reply]
You uninstalled your previous version of Rust before installing via `rustup` right? Also try to do `cargo clean` after you switched Rust installation.

**Reply by luongngocminh** [reply]
doesn't seem to change anything though. I've tried both overriding the current version, and uninstall the previous version, and cargo clean.
For context I'm using Mac if it's helpful.

**Reply by dpetka2001** [reply]
![2024-02-19_11-30](https://github.com/LazyVim/LazyVim/assets/12776461/633a8140-53ae-40f6-8146-44bbf1baf2ef)

This is what i get with a fresh Lazyvim starter installation and the Rust Extra enabled in a `cargo init` project.

Something must be wrong with your environment. Unfortunately, I don't have Mac and no experience with them.

**Reply by luongngocminh** [reply, marked as answer]
Found the culprit, the problem was the Rust LazyExtras package, some of the default LSP settings, I still don't know which, was doing something funky in the background, disable that seems to solve the problem.

**Reply by luongngocminh** [reply]
Thanks for helping.

**Reply by dpetka2001** [reply]
Strange because in my screenshot I was using Rust LazyExtras package and didn't observe any problems. Anyhow, glad you're able to solve it and go on with your work.

**Reply by luongngocminh** [reply]
Yeah, what make me realize that there is something wrong with the extras was that only error diagnostic from clippy was visible (in my case was a fixed float resembled the number Pi so clippy triggered an error, wanting to use builtin Pi constant), so I suspect that because the extras uses the `cargo clippy` check instead of `cargo check` command, and I was correct, but for some reason changing it to `cargo check` did get rid of the clippy errors, while still no diagnostic for `cargo check`, in the end I just disable it entirely.

**Reply by exor2008** [reply]
For me it was enough to remove the line

```
checkOnSave = diagnostics == "rust-analyzer"
```

from `rustaceanvim` default config.
